### PR TITLE
Split TSX from TypeScript to fix #4359

### LINKED ISSUE #4359: TSX files are identified as TypeScript rather than JSX+TS
## Preliminary Steps

I confirm I have...
- [x] reviewed [How Linguist Works](https://github.com/github/linguist#how-linguist-works),
- [x] reviewed the [Troubleshooting](https://github.com/github/linguist#troubleshooting) docs,
- [ ] considered implementing an [override](https://github.com/github/linguist#overrides), _not applicable_
- [x] verified an issue has not already been logged for your issue ([linguist issues](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+repo%3Agithub/linguist)).

## Problem Description

Files with the extension `.tsx` (and code blocks with language identifier `tsx`) on GitHub are shown as TypeScript instead of TypeScript+JSX.

### URL of the affected repository:

https://github.com/rainboxx/tsx-jsx-repro/tree/master

### Last modified on:

2018-12-21

### Expected language:

JSX, although I'm not sure whether there needs a new language that combines TS+JSX

### Detected language:

TypeScript

### PR DESCRIPTION
**Pull request by Alhadis**
This fixes the syntax highlighting errors with TSX (TypeScript with JSX/React), as described in #4359.

Note the tests are failing because of the classifier's sudden confusion between JavaScript/TypeScript… caused by `node` being listed as an interpreter for both. The separation of the TSX fixtures were obviously screwing with it.

Merging #4470 will fix this; I merged both of my branches locally, and the tests passed. So get #4470 looked at before tending to this. =)

**Template removed as it doesn't really apply.**

### CODE REVIEW

**Comment by lildude** [comment]
You got conflicts Sir 🎩

**Comment by Alhadis** [comment]
TSX and TSQL happened to fall RIGHT on top of each other. 😂 What were the odds.

Conflicts fixed!

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -4938,6 +4938,16 @@ TOML:
   codemirror_mode: toml
   codemirror_mime_type: text/x-toml
   language_id: 365
+TSX:
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Travis CI builds are failing because TSX should come after TSQL.

Comment (Alhadis) [reply lib/linguist/languages.yml]: Oops. You're right. Silly me.

**Pull request merged**